Sources/PrimerSDK/Classes/Core/Primer/Primer.swift

```diff
@@ -200,6 +210,117 @@ public class Primer {
             self?.root?.dismiss(animated: true, completion: nil)
         }
     }
+    // swiftlint:disable cyclomatic_complexity
+    public func performThreeDS(paymentMethod: PaymentMethodToken, completion: @escaping (Error?) -> Void) {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        
+        guard let clientToken = state.decodedClientToken else {
+            return completion(PrimerError.vaultFetchFailed)
+        }
+        
+        let service = ThreeDSecureService()
+        service.initializeSDK { (initResult) in
+            switch initResult {
+            case .success:
+                service.verifyWarnings { (verifyResult) in
```

Comment:
[Feedback]
This strongly ties to 3DS service the Netcetera's implementation. I'm personally not a big fan of that because we plan to have our own implementation at some point.
Would it be possible to have `verifyWarnings` as part of `service.initializeSDK`?

Reply:
yes, nice catch. I'll abstract more code once we finalize the decision logic.